wire-gson-support/src/main/java/com/squareup/wire/MessageTypeAdapter.java

```diff
@@ -13,161 +13,149 @@
       }
     } else {
-      gson.toJson(value, value.getClass(), out);
+      gson.toJson(value, value.javaClass, out)
     }
   }
 
-  private void emitUint64(Long value, JsonWriter out) throws IOException {
+  private fun emitUint64(value: Long, out: JsonWriter) {
     if (value < 0) {
-      BigInteger unsigned = POWER_64.add(BigInteger.valueOf(value));
-      out.value(unsigned);
+      val unsigned = POWER_64.add(BigInteger.valueOf(value))
+      out.value(unsigned)
     } else {
-      out.value(value);
+      out.value(value)
     }
   }
 
-  @SuppressWarnings("unchecked")
-  @Override public @Nullable M read(JsonReader in) throws IOException {
-    if (in.peek() == JsonToken.NULL) {
-      in.nextNull();
-      return null;
+  @Throws(IOException::class)
+  override fun read(`in`: JsonReader): M? {
+    if (`in`.peek() == JsonToken.NULL) {
+      `in`.nextNull()
+      return null
     }
 
-    TypeAdapter<JsonElement> elementAdapter = gson.getAdapter(JsonElement.class);
-    B builder = messageAdapter.newBuilder();
+    val elementAdapter = gson.getAdapter(JsonElement::class.java)
+    val builder = messageAdapter.newBuilder()
 
-    in.beginObject();
-    while (in.peek() != JsonToken.END_OBJECT) {
-      String name = in.nextName();
+    `in`.beginObject()
+    while (`in`.peek() != JsonToken.END_OBJECT) {
+      val name = `in`.nextName()
 
-      FieldBinding<M, B> fieldBinding = fieldBindings.get(name);
+      val fieldBinding = fieldBindings[name]
       if (fieldBinding == null) {
-        in.skipValue();
+        `in`.skipValue()
       } else {
-        JsonElement element = elementAdapter.read(in);
-        Object value = parseValue(fieldBinding, element);
-        fieldBinding.set(builder, value);
+        val element = elementAdapter.read(`in`)
+        val value = parseValue(fieldBinding, element)
+        fieldBinding.set(builder, value)
       }
     }
 
-    in.endObject();
-    return builder.build();
+    `in`.endObject()
+    return builder.build()
   }
 
-  private Object parseValue(FieldBinding<?, ?> fieldBinding, JsonElement element) {
-    if (fieldBinding.label.isRepeated()) {
-      if (element.isJsonNull()) {
-        return Collections.emptyList();
+  private fun parseValue(fieldBinding: FieldBinding<*, *>, element: JsonElement): Any {
+    if (fieldBinding.label.isRepeated) {
+      if (element.isJsonNull) {
+        return emptyList<Any>()
       }
 
-      Class<?> itemType = fieldBinding.singleAdapter().javaType;
+      val itemType = fieldBinding.singleAdapter().javaType
 
-      JsonArray array = element.getAsJsonArray();
-      List<Object> result = new ArrayList<>(array.size());
-      for (JsonElement item : array) {
-        result.add(gson.fromJson(item, itemType));
+      val array = element.asJsonArray
+      val result = ArrayList<Any>(array.size())
+      for (item in array) {
+        result.add(gson.fromJson(item, itemType))
       }
-      return result;
+      return result
```

Comment:
`return array.map { gson.fromJson(it, itemType) }`?

Reply:
WBS

Reply:
Fixed